### PR TITLE
Add support for select_last_index attribute for ArgMax & ArgMin

### PR DESCRIPTION
**Pull request by gyulaz-htec**
Onnx opset 13 intrudeced [select_last_index](https://github.com/onnx/onnx/blob/main/docs/Operators.md#attributes-1) attribute to ArgMax and ArgMin. Added support to that.

This PR resolves https://github.com/migraphx-benchmark/AMDMIGraphX/issues/81 and https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/issues/556

### CODE REVIEW

File: src/include/migraphx/op/argmax.hpp [hunk 1]
```diff
@@ -87,6 +89,10 @@ struct argmax
                 max_val   = cur_val;
                 max_index = i;
             }
+            if(select_last_index and float_equal(max_val, cur_val))
```
Comment (umangyadav) [inline src/include/migraphx/op/argmax.hpp:92]:
```suggestion
            else if(select_last_index and float_equal(max_val, cur_val))
```

IMO, doesn't make a difference but makes it more clear.

File: src/include/migraphx/op/argmin.hpp [hunk 2]
```diff
@@ -78,6 +80,10 @@ struct argmin
                 min_val   = cur_val;
                 min_index = i;
             }
+            if(select_last_index and float_equal(min_val, cur_val))
```
Comment (umangyadav) [inline src/include/migraphx/op/argmin.hpp:83]: same here

File: test/ref/argmax.cpp [hunk 3]
```diff
@@ -147,3 +147,39 @@ TEST_CASE(argmax_test_nonstd_shape)
...
+    std::vector<int64_t> res_gold = {2, 2};
+    migraphx::shape data_shape{migraphx::shape::float_type, {2, 3}};
+    auto dl = mm->add_literal(migraphx::literal{data_shape, data});
+    mm->add_instruction(migraphx::make_op("argmax", {{"axis", 1}, {"select_last_index", true}}),
+                        dl);
+    p.compile(migraphx::make_target("ref"));
+    auto result = p.eval({}).back();
+    std::vector<int64_t> result_vec;
+    result.visit([&](auto output) { result_vec.assign(output.begin(), output.end()); });
+    std::cout << std::endl;
```
Comment (umangyadav) [inline test/ref/argmax.cpp:165]:
```suggestion
```

File: test/ref/argmax.cpp [hunk 4]
```diff
@@ -147,3 +147,39 @@ TEST_CASE(argmax_test_nonstd_shape)
...
+    std::vector<int64_t> res_gold = {0, 1};
+    migraphx::shape data_shape{migraphx::shape::float_type, {2, 3}};
+    auto dl = mm->add_literal(migraphx::literal{data_shape, data});
+    mm->add_instruction(migraphx::make_op("argmax", {{"axis", 1}, {"select_last_index", false}}),
+                        dl);
+    p.compile(migraphx::make_target("ref"));
+    auto result = p.eval({}).back();
+    std::vector<int64_t> result_vec;
+    result.visit([&](auto output) { result_vec.assign(output.begin(), output.end()); });
+    std::cout << std::endl;
```
Comment (umangyadav) [inline test/ref/argmax.cpp:183]:
```suggestion
```

File: test/ref/argmin.cpp [hunk 5]
```diff
@@ -125,3 +125,39 @@ TEST_CASE(argmin_test_nonstd_shape)
...
+    std::vector<int64_t> res_gold = {2, 2};
+    migraphx::shape data_shape{migraphx::shape::float_type, {2, 3}};
+    auto dl = mm->add_literal(migraphx::literal{data_shape, data});
+    mm->add_instruction(migraphx::make_op("argmin", {{"axis", 1}, {"select_last_index", true}}),
+                        dl);
+    p.compile(migraphx::make_target("ref"));
+    auto result = p.eval({}).back();
+    std::vector<int64_t> result_vec;
+    result.visit([&](auto output) { result_vec.assign(output.begin(), output.end()); });
+    std::cout << std::endl;
```
Comment (umangyadav) [inline test/ref/argmin.cpp:143]:
```suggestion
```

File: test/ref/argmin.cpp [hunk 6]
```diff
@@ -125,3 +125,39 @@ TEST_CASE(argmin_test_nonstd_shape)
...
+    std::vector<int64_t> res_gold = {0, 1};
+    migraphx::shape data_shape{migraphx::shape::float_type, {2, 3}};
+    auto dl = mm->add_literal(migraphx::literal{data_shape, data});
+    mm->add_instruction(migraphx::make_op("argmin", {{"axis", 1}, {"select_last_index", false}}),
+                        dl);
+    p.compile(migraphx::make_target("ref"));
+    auto result = p.eval({}).back();
+    std::vector<int64_t> result_vec;
+    result.visit([&](auto output) { result_vec.assign(output.begin(), output.end()); });
+    std::cout << std::endl;
```
Comment (umangyadav) [inline test/ref/argmin.cpp:161]:
```suggestion
```

**Comment by gyulaz-htec** [comment]
Thanks @umangyadav, I've adressed you change requests.

**Review by umangyadav — APPROVED** [review]

**Pull request merged**